### ISSUE TITLE
QUESTION: Opening a metamask notification(like confirm the transaction) without modifying internal state

### ISSUE THREAD

**Issue by vkomodey** [MetaMask/metamask-extension#10688]
Hi MetaMask team! I am preparing a feature pull request for the Metamask extension. I will describe the idea and motivation more deeply in the PR later

**Context**:
I need to open a notification window during the external API call. This window includes a component that includes a captcha widget which the user should solve in order to send the result back to the background.

**Problem**: The question is - how to open a window with custom page without modifying internal state.. Ideally, I'd like to make an explicit call inside the background to show the captcha page. Like:
```
const options = { captchaRequested: true };
triggerUi({ ...windowOptions, customOptions: options });
```

TLDR. I see several options here:
1. Create a separate view page(like captcha.html) + different environment type like CAPTCHA_PAGE
I thought to use a different page(not `notification.html` here), but it sounds like a hack, since you have clear env. type segregation(in app/scripts/lib/utils.js getEnvironmentType fn) which is calculated using the URI of the page(currently it is `notification.html`)
2. Update state scheleton. Add property - `captchaRequested` and check it in the UI
 In the `ui/index.js` I see the next code:

```
  // if unconfirmed txs, start on txConf page
  const unapprovedTxsAll = txHelper(
    metamaskState.unapprovedTxs,
    metamaskState.unapprovedMsgs,
    metamaskState.unapprovedPersonalMsgs,
    metamaskState.unapprovedDecryptMsgs,
    metamaskState.unapprovedEncryptionPublicKeyMsgs,
    metamaskState.unapprovedTypedMessages,
    metamaskState.network,
    metamaskState.provider.chainId,
  );
  const numberOfUnapprovedTx = unapprovedTxsAll.length;
  if (numberOfUnapprovedTx > 0) {
    store.dispatch(
      actions.showConfTxPage({
        id: unapprovedTxsAll[0].id,
      }),
    );
  }
```

which makes me think, that I can manipulate the starting page of the window only by changing the state of the app which is not the best solution in my case because I should manage that state in the background by setting some kind of flag(e.g. `state.captchaRequested`) to `true` and then carefully invalidate it(that is the place where the pain starts)

Maybe you have other options on how it can be implemented? Thanks

**Comment by vkomodey** [comment]
After investigations I found the way to call notificationManager.showPopup with `notification.html#captcha` URL instead of default `notification.html` URL. Don't know, whether it is good solution or not, but it works
